docs/contributing/ux-guidelines.md

```diff
@@ -3,8 +3,119 @@ id: ux-guidelines
+
+Secondary to the primary user of the rhoas CLI (humans), machines **MUST** also be able to read the output (in appropriate commands). This is done to make it possible to write scripts that use the rhoas CLI, but it can also help the usability for humans, such as combining JSON output with `jq` to query data. Some data which is unnecessary for most cases may be omitted from the human-readable data output to improve readability, but users may also want to view the full payload.
+
+### **MUST** internationalize text strings in `stderr`
+
+The CLI internationalizes all text strings (with the exception of debug messages) to allow it to be usable in multiple languages. The only supported language as of now is English ("en").
+
+### **MUST** output a list of items to a table by default
+
+When displaying a list of items, it **MUST** be output into a table format by default. The CLI uses a standard table format and provides libraries to do so in consistency with the other tables.
+
+![List of items in table format](../../website/static/img/ux-assets/list-table-format.png)
+
+An `--output` flag **MUST** be provided to enable the user to output the list in either JSON or YAML format.
+
+![List of items in JSON format](../../website/static/img/ux-assets/list-table-format-json.png)
+
+### **SHOULD** display output on success
+
+When a command has finished executing successfully, brief text should be output to update the user.
+
+![Success output message](../../website/static/img/ux-assets/output-success-message.png)
+
+It is **RECOMMENDED** to prefix a success message with a green check mark "✔️" for Unicode compatible shells.
+
+### **SHOULD** display an error message on command fail
+
+When an error has occured during command execution a clear and concise error message **SHOULD** be printed before exiting the program.
+
+![Error output message](../../website/static/img/ux-assets/output-failure-message.png)
+
+### **MAY** use color in human-readable output where it enhances readability
+
+The use of color **MAY** be used if you feel that its additional will improve readability of the output. Avoid overuse of color and limit the range of colors used.
+
+The [`color`](https://pkg.go.dev/github.com/redhat-developer/app-services-cli@v0.37.1-beta3/pkg/core/ioutil/color) package provides a number of APIs which provided common colors according to how they should be used.
+
+### **SHOULD** allow the user to disable color output
+
+User's may need or want color output disabled. When the `NO_COLOR` enviroment variable is set then the additional of ANSI color to the output should be prevented.
+
+### **SHOULD NOT** display any animations in `stdout`
+
+Animations should only be displayed in `stderr`, as `stdout` needs to be kept for machine-readable output.
+
+### **MAY** use symbols and emoji where it makes things clearer
+
+Symbols and emojis can sometimes tell more than words can and they can also be useful to draw someone's attention to something.
```

Comment:
[Question] Style of this guide would be more informal ?

```suggestion
Symbols and emojis can sometimes tell more than words can and they can also be useful to draw someone's attention to something.
```

If we want to be more formal then we can say it shorter: `Developers are allowed to use emojis`

Reply:
Informal is okay. The formal bit is the way everything is split into RFC 2119 headings instead of just flowing into each other naturally.